indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/autoscaler/CostBasedAutoScaler.java

```diff
@@ -129,46 +125,25 @@ public void reset()
     // No-op.
   }
 
-  private CostMetrics collectMetrics()
+  @Override
+  public int computeTaskCountForRollover()
   {
-    if (spec.isSuspended()) {
-      log.debug("Supervisor [%s] is suspended, skipping a metrics collection", supervisorId);
-      return null;
-    }
-
-    final LagStats lagStats = supervisor.computeLagStats();
-    if (lagStats == null) {
-      log.debug("Lag stats unavailable for supervisorId [%s], skipping collection", supervisorId);
-      return null;
-    }
-
-    final int currentTaskCount = supervisor.getIoConfig().getTaskCount();
-    final int partitionCount = supervisor.getPartitionCount();
-
-    final Map<String, Map<String, Object>> taskStats = supervisor.getStats();
-    final double movingAvgRate = extractMovingAverage(taskStats, DropwizardRowIngestionMeters.ONE_MINUTE_NAME);
-    final double pollIdleRatio = extractPollIdleRatio(taskStats);
+    return computeOptimalTaskCount(collectMetrics());
```

Comment:
I wonder if for this method we shouldn't just reuse the metrics collected in the last cycle.
Metrics collection may be slow since the supervisor might need to contact all the running tasks.
This would slow down the task rollover process causing ingestion lag.

Reply:
Good idea, fixed